Without the public constructor, the class is effectively internal + @discardableResults + configuring queue

migueldeicaza opened the pull request:
This brings a constructor to the Environment class, without it, it is not possible to instantiate instances of it.

It also flags the various fluid-like APIs to be @discardableResults, to prevent errors from being raised when not using without a terminator.

Surfaces a way to configure the queue to be used for dispatching messages.

migueldeicaza committed:
Without the public constructor, the class is effectively internal

migueldeicaza committed:
Make the results of the fluid-like APIs to be discardable, to prevent warnings from calling code

migueldeicaza committed:
Allow for the callback queue to be set

migueldeicaza committed:
Allow for logging to be specified

migueldeicaza committed:
Upgrade libssh2 to 1.9.0

migueldeicaza committed:
Backport fix for timeout from github issues

migueldeicaza committed:
Upgrade libssh2

migueldeicaza committed:
Add my build script

bes commented:
@migueldeicaza I feel like using this PR / your fork is the "right thing to do" currently, until these changes are merged into master. Thanks for sharing it!

migueldeicaza committed:
Update README

migueldeicaza commented:
Of course, you can feel free to use my branch.

migueldeicaza committed:
Build an XCFramework

migueldeicaza committed:
Do not surface LibSSH2 to the swiftmodule, this fixes the 'no such module' error I get when consuming the library

migueldeicaza committed:
Use SwiftPM

migueldeicaza committed:
Add

migueldeicaza committed:
Experimentation

migueldeicaza committed:
Add new authentication system: byCallback, to allow me to wire up the secure enclave in SwiftTermApp

migueldeicaza committed:
Update README.md

migueldeicaza committed:
Make this compile with older iOS, at the expense of no support for this feature

migueldeicaza committed:
Fix the build for macOS and SwiftPM

migueldeicaza committed:
Add support for known hosts

migueldeicaza committed:
Add Fingerprint SHA256 type

migueldeicaza committed:
No human should have lost as much time as I did when I introduced this typo

migueldeicaza committed:
Session.fingerprintBytes, new methods that returns an array of bytes, rather than a formatted string

migueldeicaza committed:
Disable bitcode to work with the new SSH libraries

migueldeicaza committed:
Forgo to commit this

migueldeicaza committed:
Use my copy

migueldeicaza committed:
Fix warnings

migueldeicaza committed:
Create swift.yml

migueldeicaza committed:
Remove warning

migueldeicaza committed:
Add additional information when available on error returns

migueldeicaza committed:
Allows the session to be reused for more than one task at a time.

This enables for instance to start an SSH connection, and then
use various channels over it.

The SSHChannel no longer is a subclass of an SSHSession, which made
things awkward when attempting to use the session again.

Instead, the session is a field in the channel, and new channels can
be constructed from the session if this is available.

This also turns off the sample that defaulted to publicKey
authentication.
